torchgeo/models/cnnlstm.py

```diff
@@ -0,0 +1,378 @@
+            bidirectional: If True, becomes a bidirectional LSTM.
+            dropout: Dropout probability for LSTM layers (except last).
+            batch_first: If True, input and output tensors are provided
+                as (batch, seq, feature).
+        """
+        super().__init__()
+        self.input_dim = input_dim
+        self.hidden_dim = hidden_dim
+        self.num_layers = num_layers
+        self.bidirectional = bidirectional
+        self.batch_first = batch_first
+
+        self.lstm = nn.LSTM(
+            input_size=input_dim,
+            hidden_size=hidden_dim,
+            num_layers=num_layers,
+            bidirectional=bidirectional,
+            dropout=dropout if num_layers > 1 else 0.0,
+            batch_first=batch_first,
+        )
+
+        self.output_dim = hidden_dim * (2 if bidirectional else 1)
+
+    def forward(
+        self, x: torch.Tensor, lengths: torch.Tensor | None = None
+    ) -> torch.Tensor:
+        """Forward pass through LSTM encoder.
+
+        Args:
+            x: Input tensor of shape (B, T, input_dim).
+            lengths: Lengths of sequences for packing (optional).
+
+        Returns:
+            Output features from the LSTM of shape (B, T, hidden_dim * num_directions).
+        """
+        if lengths is not None:
+            # Pack for variable-length sequences
+            packed_x = nn.utils.rnn.pack_padded_sequence(
+                x, lengths.cpu(), batch_first=self.batch_first, enforce_sorted=False
+            )
+            packed_output, _ = self.lstm(packed_x)
+            output, _ = nn.utils.rnn.pad_packed_sequence(
+                packed_output, batch_first=self.batch_first
+            )
+        else:
+            output, _ = self.lstm(x)
+        return output
+
+
+class Task(nn.Module):
+    """Unified task head supporting classification and regression.
+
+    Args:
+        input_dim (int): Input feature dimension.
+        task_type (str): Task type, either 'classification' or 'regression'.
+        num_classes (int, optional): Number of classes for classification.
+            Required if task_type='classification'.
+        output_dim (int): Output dimension for regression (default 1).
+        dropout (float): Dropout rate applied before the output layer.
+        pooling (str): Method to pool sequence inputs into a single vector.
+            Options: 'last', 'mean', 'max', 'attention'.
+
+    Forward:
+        x (torch.Tensor): Input tensor, either:
+            - (B, input_dim) for single vector, or
+            - (B, T, input_dim) for sequences.
+
+    Returns:
+        Dict[str, Optional[torch.Tensor]]:
+            For classification:
+                - 'outputs': Raw logits tensor (B, num_classes).
+                - 'probs': Softmax probabilities tensor (B, num_classes).
+            For regression:
+                - 'outputs': Regression predictions (B, output_dim).
+                - 'probs': None.
```

Comment:
classification/regression are the same thing, the only difference is the `num_classes`. Both should return raw logits, no probabilities.

Not sure if it makes sense to use this model for semantic segmentation or pixelwise regression. Maybe the `Decoder` should just have a `num_classes` parameter and no `Task` is needed?